### PR TITLE
Addition of a Password to the Homepage

### PR DESCRIPTION
**Pull request by Adebesin-Cell**
### Changes.
- Added login page for pre-release of ``IQ-GPT``
- Redirect unauthorized users.
- Added baseURL flow.

###Media

pika-1688545505204-1x

pika-1688547087065-1x

### Linked Issues:

Fixes https://github.com/EveripediaNetwork/issues/issues/1508

### CODE REVIEW

**Review by Royal-lobster — APPROVED** [review]

**Comment by kesar** [comment]
Encrypt? 👀  😆

File: src/env.mjs [hunk 1]
```diff
@@ -27,5 +28,6 @@ export const env = createEnv({
     BIGQUERY_KEY: process.env.BIGQUERY_KEY,
     COINGECKO_API_KEY: process.env.COINGECKO_API_KEY,
     BLOCKCHAIR_API_KEY: process.env.BLOCKCHAIR_API_KEY,
+    NEXT_PUBLIC_APP_PASSWORD: process.env.NEXT_PUBLIC_APP_PASSWORD,
```
Comment (kesar) [inline src/env.mjs:31]: i doubt anyone is gonna try to hack this, but if you do NEXT_PUBLIC_APP_PASSWORD the password gets leaked in the UI :)

**Comment by Adebesin-Cell** [comment]
> Encrypt? 👀 😆

😂 I went with the first thing that came to my mind.

File: src/env.mjs (hunk 1 above)
Comment (Adebesin-Cell) [reply src/env.mjs:31]: Noted, will make add the fix now.

Comment (kesar) [reply src/env.mjs:31]: Captura de pantalla 2023-07-05 a las 11 29 58

Comment (Adebesin-Cell) [reply src/env.mjs:31]: Can you please clarify ser?

Comment (kesar) [reply src/env.mjs:31]: just remove the next public, screenshot was about showing how next public can be accessed from browser

Comment (Adebesin-Cell) [reply src/env.mjs:31]: Oh, alright ser. Thanks!

**Review by Simplypheyie96 — APPROVED** [review]

**Pull request merged**